### PR TITLE
Aggregate all modules in root

### LINKED ISSUE #310: Add CI steps to check all sources are compilable
Currently (at least) `example` and `collector` modules are not checked until some late step in CI on master.

### PR DESCRIPTION
**Pull request by catostrophe**
Closes #310

### CODE REVIEW

File: modules/example/src/main/scala/io/janstenpickle/trace4cats/example/Fs2Example.scala [hunk 1]
```diff
@@ -62,9 +62,12 @@ object Fs2Example extends IOApp {
   def doWork[F[_]: BracketThrow: Clock](stream: TracedStream[F, FiniteDuration]): TracedStream[F, Long] =
     stream
       // eval some effect within a span
-      .evalMap("this is child of the initial injected root span", SpanKind.Internal, "optional-attribute" -> true) {
-        dur =>
-          Clock[F].realTime(TimeUnit.MILLISECONDS).map(_ + dur.toMillis)
+      .evalMap(
+        "this is child of the initial injected root span",
+        SpanKind.Internal,
+        "optional-attribute" -> BooleanValue(true)
```
Comment (catostrophe) [inline modules/example/src/main/scala/io/janstenpickle/trace4cats/example/Fs2Example.scala:68]: to compile on scala 2.12

File: README.md [hunk 2]
```diff
@@ -88,18 +88,20 @@ import cats.implicits._
...
+
 import scala.concurrent.duration._
 
 object Trace4CatsQuickStart extends IOApp {
   def entryPoint[F[_]: Concurrent: ContextShift: Timer: Logger](
     blocker: Blocker,
     process: TraceProcess
   ): Resource[F, EntryPoint[F]] =
-    AvroSpanCompleter.udp[F](blocker, process, batchTimeout = 50.millis).map { completer =>
+    AvroSpanCompleter.udp[F](blocker, process, config = CompleterConfig(batchTimeout = 50.millis)).map { completer =>
```
Comment (catostrophe) [inline README.md:104]: fix the doc

File: build.sbt [hunk 3]
```diff
@@ -314,7 +321,6 @@ lazy val `opentelemetry-common` =
 lazy val `opentelemetry-jaeger-exporter` =
   (project in file("modules/opentelemetry-jaeger-exporter"))
     .settings(publishSettings)
-    .settings(commonSettings)
```
Comment (catostrophe) [inline build.sbt:317]: this is redundant

**Review by janstenpickle — APPROVED** [review]

**Comment by catostrophe** [comment]
@janstenpickle and please approve this one. would be helpful for refactoring since it checks all the code

**Pull request merged**